### DISCUSSION TITLE
How to export annotations to COCO format or similar to refine annotations?

### DISCUSSION THREAD

**Discussion by Xavier-Waters** [deepdoctection/deepdoctection#347] [Q&A]
Hello I am currently working a on a project using this project and am wondering how to export the annotations created by page.save() to a standard format so I can use the model to create more annotations when I have a model that is 90% there and just needs a little refinement, I looked all over the docs and the internet but could not find anything.

**Comment by JaMe76** [comment]
There is a `image_to_coco` function that converts the `Image` data structure into a COCO-format. I haven't used this function for a while, it's a kind of legacy code, but it will get you started.

Disclaimer: The following code snippets will give you an idea of how to do it. I haven't tested it and might require some (slight) modifications.

```python
import deepdoctection as dd

df = dd.SerializerFiles.load(path="your/path/to/json-dir") # loading a path to your json-file
df = dd.MapData(df , dd.Image.from_file) # loading the .json file and converting into the deepdoctection format
df = dd.MapData(df, dd.image_to_coco) # converting the image object into a COCO-snippet. If this function is not available under this      namespace, it has been implemented in `deepdoctection.mapper.cocostruct`

df.reset_state()

coco_format = {"info": {...},   # your info to be specified
                           "licenses": [...], # your license to be specified
                            "images": [],
                            "annotations": [],
                            "categories": []}. # your categories to be specified
for img, anns in df:    #    image_to_coco returns a tuple: img: dict, anns: list[dict] that must be collected to generate the COCO-structure.  
   coco_format["images"].append(img)
   coco_format["annotations"].append(anns)

# finally saving coco_format to JSON.

**Reply by JaMe76** [reply]
But you could also setup a Dataset that streams the annotations in deepdoctection format directly to the model training. There are some discussions about that [here](https://github.com/deepdoctection/deepdoctection/discussions/42). You can then use the custom Tensorpack or Detectron2 training script to improve your layout model. There is a similar (more advanced) tutorial [here](https://deepdoctection.readthedocs.io/en/latest/tutorials/training_and_logging/).
